flink-ml-lib/src/main/java/org/apache/flink/ml/classification/logisticregression/OnlineLogisticRegression.java

```diff
@@ -0,0 +1,428 @@
+                }
+                double p = 0.0;
+                for (int i = 0; i < indices.length; ++i) {
+                    int idx = indices[i];
+                    if (Math.abs(zParam[idx]) <= l1) {
+                        weights[idx] = 0.0;
+                    } else {
+                        weights[idx] =
+                                ((zParam[idx] < 0 ? -1 : 1) * l1 - zParam[idx])
+                                        / ((beta + Math.sqrt(nParam[idx])) / alpha + l2);
+                    }
+                    p += weights[idx] * values[i];
+                }
+                p = 1 / (1 + Math.exp(-p));
+                for (int i = 0; i < indices.length; ++i) {
+                    int idx = indices[i];
+                    double g = (p - label) * values[i];
+                    double sigma =
+                            (Math.sqrt(nParam[idx] + g * g) - Math.sqrt(nParam[idx])) / alpha;
+                    zParam[idx] += g - sigma * weights[idx];
+                    nParam[idx] += g * g;
+                    weights[idx] += 1.0;
+                }
+            }
+            if (points.length > 0) {
+                output.collect(
+                        new StreamRecord<>(
+                                new DenseVector[] {
+                                    new DenseVector(weights),
+                                    new DenseVector(zParam),
+                                    new DenseVector(nParam)
+                                }));
+            }
+        }
+
+        @Override
+        public void processElement2(StreamRecord<DenseVector[]> modelDataRecord) throws Exception {
+            modelDataState.clear();
+            modelDataState.add(modelDataRecord.getValue());
+        }
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        ReadWriteUtils.saveMetadata(this, path);
+        ReadWriteUtils.saveModelData(
+                LogisticRegressionModelData.getModelDataStream(initModelDataTable),
+                path,
+                new LogisticRegressionModelData.ModelDataEncoder());
+    }
+
+    public static OnlineLogisticRegression load(StreamTableEnvironment tEnv, String path)
+            throws IOException {
+        OnlineLogisticRegression onlineLogisticRegression = ReadWriteUtils.loadStageParam(path);
+        Table modelDataTable =
+                ReadWriteUtils.loadModelData(
+                        tEnv, path, new LogisticRegressionModelData.ModelDataDecoder());
+        onlineLogisticRegression.setInitialModelData(modelDataTable);
+        return onlineLogisticRegression;
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    /**
+     * An operator that splits a global batch into evenly-sized local batches, and distributes them
+     * to downstream operator.
+     */
+    private static class GlobalBatchSplitter implements FlatMapFunction<Row[], Row[]> {
```

Comment:
Can we reuse the code snippet in `OnlineKmeans`, rather than copy-paste it here?